woocommerce/woocommerce
# Add remote logger as a log handler to wc logger

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR address a comment from https://github.com/woocommerce/woocommerce/pull/49599#pullrequestreview-2189423877, suggesting that RemoteLogger could be added as a log handler to the wc logger so we can potentially log more than just fatal errors in the future.

Changes:

- Add remote logger as a log handler to wc logger
- Split handle() code into two methods: should_handle() and log() to make it easier to understand
- Ignore log entries that are not critical (info, debug, etc)
- Improve and add more tests




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a JN site with this branch
2. Upload and replace woocommerce-beta-tester.zip [woocommerce-beta-tester.zip](https://github.com/user-attachments/files/16519743/woocommerce-beta-tester.zip)
3. Go to `Tools -> WCA Test Helper -> Remote Logging`
4. Click on PHP Simulate Core Exception, refresh the page and confirm it throws an error
5. Check the log server page for the log entry (See test plan in D155228-code to see how to check the log entry)
6. Click on PHP `Reset Rate Limit`, and click on PHP `Log Event` and confirm the log entry is created




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Add remote logger as a log handler to wc logger

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
